### PR TITLE
add_layer_from_images: allow single image folders, retry with bioformats

### PR DESCRIPTION
**Pull request by jstriebel**
### Description:
Improvments for `ds.add_layer_from_images`:
* Always retry with bioformats if other options fail (part of #826)
* Expand files explicitly, allowing to supply a path to a folder which contains a single image (fixes bug in #827)

### Todos:
 - [x] Updated Changelog
 - [x] Updated Documentation
 - [x] Added / Updated Tests
 - [x] Considered adding this to the Examples --> see #828

### CODE REVIEW

File: webknossos/webknossos/dataset/_utils/pims_images.py [hunk 1]
```diff
@@ -290,48 +290,85 @@ def _open_images(
...
+                            images_context_manager = pims.ReaderSequence(
+                                original_images, pims.bioformats.BioformatsReader
+                            )
+                        else:
+                            images_context_manager = pims.bioformats.BioformatsReader(
+                                original_images
+                            )
+                    except Exception as e2:
+                        if len(pims_open_exceptions) == 0:
+                            raise
```
Comment (philippotto) [inline webknossos/webknossos/dataset/_utils/pims_images.py:361]:
```suggestion
                            raise e2
```

I find that clearer.

File: webknossos/webknossos/dataset/_utils/pims_images.py [hunk 2]
```diff
@@ -290,48 +290,85 @@ def _open_images(
...
+                if isinstance(original_images, Path):
+                    original_images = str(original_images)
                 else:
-                    original_images = self._original_images
-                    if isinstance(original_images, Path):
-                        original_images = str(original_images)
+                    try:
+                        if isinstance(original_images[0], Path):
+                            original_images = [str(i) for i in original_images]
+                    except Exception as e2:
```
Comment (philippotto) [inline webknossos/webknossos/dataset/_utils/pims_images.py:315]: there is no e1 in this method, is there?

File: webknossos/tests/dataset/test_add_layer_from_images.py [hunk 3]
```diff
@@ -75,6 +75,16 @@ def test_compare_tifffile(tmp_path: Path) -> None:
         1,
         (1024, 1024, 12),
     ),
+    (
+        "testdata/temca2",
```
Comment (philippotto) [inline webknossos/tests/dataset/test_add_layer_from_images.py:79]: temca2 contains multiple image files, right?

**Review by philippotto — COMMENTED** [review]
Looks good to me! What do you think about splitting up the `_open_images` method to make it a bit more readable?

File: webknossos/tests/dataset/test_add_layer_from_images.py (hunk 3 above)
Comment (jstriebel) [reply webknossos/tests/dataset/test_add_layer_from_images.py:79]: Yes :+1:

**Comment by jstriebel** [comment]
> What do you think about splitting up the _open_images method to make it a bit more readable?

Good idea, did that and numbered the errors correctly :+1:

**Review by philippotto — APPROVED** [review]
LGTM :) :100:

**Pull request merged**